syntax = "proto3";

service AlgoService{

    rpc getSingleModalData (SingleModalRequest) returns (SingleModalResponse) {}

    rpc getMultiModalData (MultiModalRequest) returns (MultiModalResponse) {}
}

message SingleModalRequest{
    string type = 1;
    string singleParamter = 2;
}

message SingleModalResponse{
    string type = 1;
    string data = 2;
}

message MultiModalRequest{
    string type = 1;
    string multiParamter = 2;
}

message MultiModalResponse{
    string type = 1;
    string data = 2;
}